packages/docusaurus-plugin-openapi-docs/src/markdown/utils.ts

```diff
@@ -9,16 +9,27 @@ export type Children = string | undefined | (string | string[] | undefined)[];
 
 export type Props = Record<string, any> & { children?: Children };
 
-export function create(tag: string, props: Props): string {
+export type Options = { inline?: boolean };
+
+export function create(
+  tag: string,
+  props: Props,
+  options: Options = {}
+): string {
   const { children, ...rest } = props;
 
   let propString = "";
   for (const [key, value] of Object.entries(rest)) {
     propString += `\n  ${key}={${JSON.stringify(value)}}`;
   }
-  propString += propString ? "\n" : "";
-
   let indentedChildren = render(children).replace(/^/gm, "  ");
+
+  if (options.inline) {
+    propString += `\n  children={${JSON.stringify(children)}}`;
+    indentedChildren = "";
+  }
```

Comment:
I had to support this option because indentedChildren was generating a nested `p` inside `tag` (H1, H2) because it was breaking line

to support inline children for `tag` I had to use propString as MDX syntax is strict on that. Final output looked like:

```mdx
Petstore offers two forms of authentication:
  - API Key
  - OAuth2
  
OAuth2 - an open protocol to allow secure authorization in a simple
and standard method from web, mobile and desktop applications.


<div
  style={{"marginBottom":"2rem"}}
>
  <Heading
    id={"authentication"}
    as={"h2"}
    className={"openapi-tabs__heading"}
    children={"Authentication"}
  >
  </Heading><SchemaTabs
    className={"openapi-tabs__security-schemes"}
  >
    <TabItem
      label={"OAuth 2.0: petstore_auth"}
      value={"petstore_auth"}
    >
      
      
      Get access to data while protecting your account credentials.
      OAuth2 is also a safer and more secure way to give you access.
      

```

Reply:
Makes sense, thanks for the explanation!